mlos_bench/mlos_bench/config/services/remote/azure/scripts/setup-rg/setup-rg.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+##
+## Copyright (c) Microsoft Corporation.
+## Licensed under the MIT License.
+##
+
+set -eu
+
+# Argument parsing
+while [[ "$#" -gt 0 ]]; do
+case $1 in
+    # Main control plane ARM template params
+    --controlPlaneArmParameters)
+        controlPlaneArmParameters="$2"
+        shift 2
+        ;;
+    # Results DB ARM template params
+    --resultsDbArmParameters)
+        resultsDbArmParameters="$2"
+        shift 2
+        ;;
+    # Other params
+    --servicePrincipalName)
+        servicePrincipalName="$2"
+        shift 2
+        ;;
+    --resourceGroupName)
+        resourceGroupName="$2"
+        shift 2
+        ;;
+    --certName)
+        certName="$2"
+        shift 2
+        ;;
+    --certExpirationYears)
+        certExpirationYears="$2"
+        shift 2
+        ;;
+    *)
+        echo "Unknown parameter passed $1"
+        shift 1
+        ;;
+esac
+done
+
+# Default values
+resultsDbArmParameters=${resultsDbArmParameters:-""}
+certExpirationYears=${certExpirationYears:-1}
+
+# Provision resources into the resource group with ARM template
+echo "Provisioning control plane resources..."
+deploymentResults=$(az deployment group create \
+    --resource-group "$resourceGroupName" \
+    --template-file ./rg-template.json \
+    --parameters "$controlPlaneArmParameters" \
+    )
+
+if [[ ! $? ]]; then
```

Comment:
```suggestion
if [[ $? -ne 0 ]]; then
```

Reply:
`! $?` in bash is going to evaluate to "not 1 (true)` which is the opposite of what you want here.  Better to be explicit in this case.